content/400-reference/200-api-reference/050-prisma-client-reference.mdx

```diff
@@ -2006,14 +2006,18 @@ Sorts a list of records. See also: [Sorting](/concepts/components/prisma-client/
 
 - In [2.16.0](https://github.com/prisma/prisma/releases/2.16.0) and later, you can [order by relation fields](#sort-post-by-the-related-user-records-name) (Preview) - for example, order posts by the author's name.
 
-- In [3.5.0](https://github.com/prisma/prisma/releases/3.5.0) and later, in PostgreSQL you can [order by relevance](#sort-post-by-relevance-of-the-title). For more details, see [Sort by relevance](/concepts/components/prisma-client/filtering-and-sorting#sort-by-relevance-postgresql).
+- In [3.5.0](https://github.com/prisma/prisma/releases/3.5.0) and later, in PostgreSQL you can [order by relevance](#sort-post-by-relevance-of-the-title). For details, see [Sort by relevance](/concepts/components/prisma-client/filtering-and-sorting#sort-by-relevance-postgresql).
+
+- In [4.1.0](https://github.com/prisma/prisma/releases/4.1.0) and later, you can [sort `null` records first or last](#sort-post-by-the-related-user-records-name-with-nullrecords-first). For details, see [Sort with nulls first or last](/concepts/components/prisma-client/filtering-and-sorting#sort-with-null-records-first-or-last).
 
 #### Inputs
 
-| Name   | Description                  |
-| ------ | ---------------------------- |
-| `asc`  | Sort ascending (A &rarr; Z)  |
-| `desc` | Sort descending (Z &rarr; A) |
+| Name    | Description                                                                                                                          |
+| ------- | ------------------------------------------------------------------------------------------------------------------------------------ |
+| `asc`   | Sort ascending (A &rarr; Z)                                                                                                          |
+| `desc`  | Sort descending (Z &rarr; A)                                                                                                         |
+| `first` | Optional. Sort with `null` values first. <br></br>- For use on optional fields only.<br></br>- Available in version 4.1.0 and later) |
+| `last`  | Optional. Sort with `null` values last.<br></br>- For use on optional fields only.<br></br>- Available in version 4.1.0 and later)   |
```

Comment:
This makes it a bit weird now. `asc` and `desc` are the inputs for the `sort` argument, `first` and `last` are the inputs of the `nulls` argument.

As a recap, here are all the options possible for an optional field now:

This can be referred to as the "shorthand" syntax 👇
```js
prisma.x.findMany({ orderBy: { opt_field: "asc" } })
prisma.x.findMany({ orderBy: { opt_field: "desc" } })
```
And
```js
prisma.x.findMany({ orderBy: { opt_field: { sort: "asc" } } }) // or sort: "desc"

prisma.x.findMany({ orderBy: { opt_field: { sort: "asc", nulls: "first" } } }) // or sort: "asc", or nulls: "last"
```

Reply:
Ah OK, thanks - I'll think about how to restructure this section, then.

Reply:
I've restructured it now so it makes more sense.

Reply:
I've now restructured this.